core/state/database.go

```diff
@@ -509,3 +525,29 @@ func (db *cachingDB) AddRootTranslation(originalRoot, translatedRoot common.Hash
 func (db *cachingDB) SetLastMerkleRoot(merkleRoot common.Hash) {
 	db.LastMerkleRoot = merkleRoot
 }
+
+func (db *cachingDB) SaveTransitionState(root common.Hash) {
+	if db.TransitionStatePerRoot == nil {
+		db.TransitionStatePerRoot = make(map[common.Hash]*TransitionState)
+	}
+
+	db.TransitionStatePerRoot[root] = db.CurrentTransitionState
+}
+
+func (db *cachingDB) LoadTransitionState(root common.Hash) {
+	if db.TransitionStatePerRoot == nil {
+		db.TransitionStatePerRoot = make(map[common.Hash]*TransitionState)
+	}
+
+	ts, ok := db.TransitionStatePerRoot[root]
+	if !ok || ts == nil {
+		// Start with a fresh state
+		ts = &TransitionState{ended: db.triedb.IsVerkle()}
```

Comment:
this is where we handle the "verkle at genesis case" since if that happens, then the loaded transition state will be `nil` and so it is initialized.

Note that this hasn't been tested yet, but should be before everything is merged back into `kaustinen-post-shapella`.